apps/theming-designer/src/components/ThemingDesigner.tsx

```diff
@@ -0,0 +1,202 @@
+                placeholder="Select an Option"
+                label="Theme dropdown"
+                ariaLabel="Theme dropdown"
+                options={[{ key: 'light', text: 'Light theme' }, { key:  'dark', text: 'Dark theme' }]}
+                /> */}
+
+                {/* the three base slots, prominently displayed at the top of the page */}
+                <ThemeDesignerColorPicker
+                  color={this.state.primaryColor}
+                  onColorChange={this._onPrimaryColorPickerChange}
+                  label={'Primary theme color'}
+                />
+                <ThemeDesignerColorPicker
+                  color={this.state.textColor}
+                  onColorChange={this._onTextColorPickerChange}
+                  label={'Body text color'}
+                />
+                <ThemeDesignerColorPicker
+                  color={this.state.backgroundColor}
+                  onColorChange={this._onBkgColorPickerChange}
+                  label={'Body background color'}
+                />
+              </Stack>
+            </div>
+            <Stack className={mainCardsClassName}>
+              <ThemeProvider theme={this.state.theme}>
+                <Samples />
+              </ThemeProvider>
+              <AccessibilityChecker theme={this.state.theme} themeRules={this.state.themeRules} />
+              <FabricPalette themeRules={this.state.themeRules} />
+              <SemanticSlots />
+            </Stack>
+          </Stack>
+        </Stack>
+      </div>
+    );
+  }
+
+  private _onPrimaryColorPickerChange(newColor: IColor | undefined) {
+    this._onColorChange(this.state.primaryColor, BaseSlots.primaryColor, newColor);
+  }
+
+  private _onTextColorPickerChange(newColor: IColor | undefined) {
+    this._onColorChange(this.state.textColor, BaseSlots.foregroundColor, newColor);
+  }
+
+  private _onBkgColorPickerChange(newColor: IColor | undefined) {
+    this._onColorChange(this.state.backgroundColor, BaseSlots.backgroundColor, newColor);
+  }
+
+  private _makeNewTheme = (): void => {
+    if (this.state.themeRules) {
+      const themeAsJson: { [key: string]: string } = ThemeGenerator.getThemeAsJson(this.state.themeRules);
+
+      const finalTheme = createTheme({
+        ...{ palette: themeAsJson }
+        // isInverted: isDark(this.state.themeRules[BaseSlots[BaseSlots.backgroundColor]].color!)
+      });
+      this.setState({ theme: finalTheme });
+      // const root = document.querySelector('.app') as HTMLElement;
+      // if (root) {
+      //   root.style.backgroundColor = finalTheme.semanticColors.bodyBackground;
+      //   root.style.color = finalTheme.semanticColors.bodyText;
+      // }
+
+      // document.body.style.backgroundColor = finalTheme.semanticColors.bodyBackground;
+      // document.body.style.color = finalTheme.semanticColors.bodyText;
+    }
+  };
+
+  private _onColorChange = (colorToChange: IColor, baseSlot: BaseSlots, newColor: IColor | undefined) => {
```

Comment:
Can first check if `newColor === undefined` and return, then drop all of the `!` from the method.